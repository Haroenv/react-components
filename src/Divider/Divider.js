import React from 'react';
import { pure, skinnable, props, t } from '../utils';
import cx from 'classnames';

const orientation = t.enums.of(['horizontal', 'vertical'], 'orientation');
const sizeType = t.enums.of(['small', 'medium', 'large', 'no-margin'], 'sizeType');

const dividerProps = {
  orientation: t.maybe(orientation),
  style: t.maybe(t.Obj),
  size: t.maybe(sizeType)
};

const defaultDividerProps = {
  orientation: 'vertical',
  size: 'small',
  style: {}
};

/** A simple component used to visually divide UI elements
 * @param orientation - vertical | horizontal
 * @param size - size of margins
 */
@pure
@skinnable()
@props(dividerProps)
export default class Divider extends React.Component {

  static defaultProps = defaultDividerProps;

  getLocals() {
    return {
      ...this.props
    };
  }

  template({ orientation, style, size }) {
    return (
      <div className={cx('divider', orientation, size)} style={style} />
    );
  }
}
