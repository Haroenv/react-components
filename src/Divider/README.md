# Divider

## A simple component used to visually divide UI elements

|Name|Type|Default|Description|
|----|----|-------|-----------|
| orientation | ["horizontal","vertical"] | "vertical" | vertical - horizontal |
| style | Object | {} | inline-style overrides for wrapper element |
| size | ["small","medium","large","no-margin"] | "small" | size of margins |