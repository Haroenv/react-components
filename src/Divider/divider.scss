@import '../utils/theme/variables.scss';

$color: $brc-silver !default;

.divider {
  background-color: $color;
  color: $color;
  &.small {
    margin: 5px;
  }
  &.medium {
    margin: 10px;
  }
  &.large {
    margin: 20px;
  }
  &.no-margin {
    margin: 0;
  }
  &.horizontal {
    display: inline;
    margin-top: 0;
    margin-bottom: 0;
    min-height: 100%;
    border-left: 1px solid;
  }
  &.vertical {
    display: block;
    margin-left: 0;
    margin-right: 0;
    min-width: 100%;
    height: 1px;
  }
}
