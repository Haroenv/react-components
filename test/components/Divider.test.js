import React from 'react';
import render from '../render';

import Divider from '../../src/Divider';

describe('Divider', () => {

  it('renders correctly vertical', () => {
    const tree = render(
      <Divider orientation='vertical' />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly horizontal', () => {
    const tree = render(
      <Divider orientation='horizontal' />
    );
    expect(tree).toMatchSnapshot();
  });

});
