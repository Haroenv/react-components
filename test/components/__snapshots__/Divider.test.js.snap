exports[`Divider renders correctly horizontal 1`] = `"<div className=\"divider horizontal small\" style={{...}} />"`;

exports[`Divider renders correctly vertical 1`] = `"<div className=\"divider vertical small\" style={{...}} />"`;
